### PR TITLE
Buy sell marketplace pre at blobber

### PR DESCRIPTION
**Pull request by Is2PidGuy**
### **For Download**

download

table

### CODE REVIEW

File: code/go/0chain.net/blobbercore/handler/object_operation_handler.go [hunk 1]
```diff
@@ -401,6 +404,58 @@ func (fsh *StorageHandler) DownloadFile(ctx context.Context, r *http.Request) (
 	var response = &DownloadResponse{}
 	response.Success = true
 	response.LatestRM = readMarker
+	if attrs.PreAtBlobber {
+		// buyerPublicKey := r.FormValue("public_key")
+		buyerEncryptionPublicKey := "qCj3sXXeXUAByi1ERIbcfXzWN75dyocYzyRXnkStXio="
```
Comment (Is2PidGuy) [inline code/go/0chain.net/blobbercore/handler/object_operation_handler.go:409]: Need to fetch it from the shareinfo table.

File: sql/14-add-marketplace-table.sql [hunk 2]
```diff
@@ -0,0 +1,22 @@
+\connect blobber_meta;
```
Comment (moldis) [inline sql/14-add-marketplace-table.sql:1]: You might consider using auto migration with gorm instead of manually generating tables

Comment (Is2PidGuy) [reply sql/14-add-marketplace-table.sql:1]: Will definitely take a look to find which is beetter

**Review by Sriep — CHANGES_REQUESTED** [review]
Lint errors to fix https://github.com/0chain/blobber/pull/135/checks?check_run_id=3010034518

**Review by Sriep — APPROVED** [review]

**Pull request merged**